src/ReturnTypes/Helpers/AppExtension.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+declare(strict_types=1);
+
+namespace NunoMaduro\Larastan\ReturnTypes\Helpers;
+
+use Illuminate\Foundation\Application;
+use NunoMaduro\Larastan\Concerns\HasContainer;
+use PhpParser\Node\Expr;
+use PhpParser\Node\Expr\ClassConstFetch;
+use PhpParser\Node\Expr\FuncCall;
+use PhpParser\Node\Name\FullyQualified;
+use PhpParser\Node\Scalar\String_;
+use PHPStan\Analyser\Scope;
+use PHPStan\Reflection\FunctionReflection;
+use PHPStan\Type\DynamicFunctionReturnTypeExtension;
+use PHPStan\Type\ErrorType;
+use PHPStan\Type\MixedType;
+use PHPStan\Type\ObjectType;
+use PHPStan\Type\Type;
+use Throwable;
+
+class AppExtension implements DynamicFunctionReturnTypeExtension
+{
+    use HasContainer;
+
+    public function isFunctionSupported(FunctionReflection $functionReflection): bool
+    {
+        return $functionReflection->getName() === 'app';
+    }
+
+    public function getTypeFromFunctionCall(
+        FunctionReflection $functionReflection,
+        FuncCall $functionCall,
+        Scope $scope
+    ): Type {
+        if (count($functionCall->args) === 0) {
+            return new ObjectType(Application::class);
+        }
+
+        /** @var Expr $expr */
+        $expr = $functionCall->args[0]->value;
+
+        if ($expr instanceof String_) {
```

Comment:
Why only strings? What about `app(1)`? I know it is a silly example. But I think we should try to resolve everything out of the container and handle the error cases (returning null or throwing exceptions) This also mimics the actual implementation of the `app` better.

Reply:
I am check the type, since `$this->resolve` takes a string, and `Expr` does not have a toString method. If there is a simple way I can pass any `Expr` to the resolve method, please let me know.

Reply:
Yeah, sorry. You are right. This looks correct. :+1: